File: AccuPay/Views/Time Attendance Form/Overtimes/OvertimeRowRecord.vb
Option Strict On

Imports AccuPay.Core.Entities
Imports AccuPay.Core.Interfaces.Excel
Imports AccuPay.Utilities.Attributes

Public Class OvertimeRowRecord
    Inherits ExcelEmployeeRowRecord
    Implements IExcelRowRecord

    <Ignore>
    Public Property EmployeeFullName As String

    <ColumnName("Employee ID")>
    Public Property EmployeeNumber As String

    <ColumnName("Effective start date")>
    Public Property StartDate As Date?

    <ColumnName("Effective Start Time")>
    Public Property StartTime As TimeSpan?

    <ColumnName("Effective End Time")>
    Public Property EndTime As TimeSpan?

    <Ignore>
    Public Property ErrorMessage As String

    <Ignore>
    Public Property LineNumber As Integer Implements IExcelRowRecord.LineNumber

    Public Function ToOvertime(employee As Employee) As Overtime
        Dim employeeId = employee.RowID.Value
        Dim organizationId = employee.OrganizationID.Value

        If StartDate Is Nothing Then Return Nothing

        Return Overtime.NewOvertime(
            organizationId:=organizationId,
            employeeId:=employeeId,
            startDate:=StartDate.Value,
            startTime:=StartTime,
            endTime:=EndTime)

    End Function

End Class

File: AccuPay.Core/Services/Imports/Overtimes/OvertimeRowRecord.cs
using AccuPay.Core.Interfaces.Excel;
using AccuPay.Utilities.Attributes;
using System;

namespace AccuPay.Core.Services.Imports.Overtimes
{
    public class OvertimeRowRecord : ExcelEmployeeRowRecord, IExcelRowRecord
    {
        public int LineNumber { get; set; }

        [ColumnName("EmployeeID")]
        public string EmployeeNo { get; set; }

        [ColumnName("Effective start date")]
        public DateTime? StartDate { get; set; }

        [ColumnName("Effective Start Time")]
        public DateTime? StartTime { get; set; }

        [ColumnName("Effective End Time")]
        public DateTime? EndTime { get; set; }
    }
}
